core/src/main/scala/magnolia1/magnolia.scala

```diff
@@ -640,29 +639,50 @@ object Magnolia {
             .fold(error(_), identity)
         }
 
-        val assignments = typeclasses.zipWithIndex.map { case ((subType, typeclass), idx) =>
-          val symbol = subType.typeSymbol
-          val (annotationTrees, inheritedAnnotationTrees) = annotationsOf(symbol)
-          q"""$subtypesVal($idx) = $SubtypeObj[$typeConstructor, $genericType, $subType](
-              ${typeNameOf(subType)},
-              $idx,
-              $ArrayObj(..${annotationTrees}): Array[_root_.scala.Any],
-              $ArrayObj(..${inheritedAnnotationTrees}): Array[_root_.scala.Any],
-              $ArrayObj(..${typeAnnotationsOf(symbol, fromParents = true)}): Array[_root_.scala.Any],
-              $CallByNeedObj($typeclass),
-              (t: $genericType) => t.isInstanceOf[$subType],
-              (t: $genericType) => t.asInstanceOf[$subType]
-            )"""
-        }
-
         val subType = appliedType(SubtypeTpe, typeConstructor, genericType)
+        val groupSize = 500
+        val partialSubtypesVal = c.freshName(TermName("partialSubtypes"))
+        val (functionNames, partialAssignmentFunctions) = typeclasses
+          .grouped(groupSize)
+          .toList
+          .map { group =>
+            val name = c.freshName(TermName("partialAssignments"))
+            val assignments = group.zipWithIndex.map { case ((subType, typeclass), idx) =>
+              val symbol = subType.typeSymbol
+              val (annotationTrees, inheritedAnnotationTrees) = annotationsOf(symbol)
+              q"""$partialSubtypesVal($idx) = $SubtypeObj[$typeConstructor, $genericType, $subType](
+                        ${typeNameOf(subType)},
+                        $idx,
+                        $ArrayObj(..${annotationTrees}): Array[_root_.scala.Any],
+                        $ArrayObj(..${inheritedAnnotationTrees}): Array[_root_.scala.Any],
+                        $ArrayObj(..${typeAnnotationsOf(symbol, fromParents = true)}): Array[_root_.scala.Any],
+                        $CallByNeedObj($typeclass),
+                        (t: $genericType) => t.isInstanceOf[$subType],
+                        (t: $genericType) => t.asInstanceOf[$subType]
```

Comment:
You can extract those as helper methods in the beginning to reduce the method size

Reply:
Ah, I understand what you mean. Yes, it seems we need to generate sort of runtime reflection codes by macro. So I would say keep the current way is better even if it reduce the method size dramatically. What do you think?

Reply:
Yes, let's keep it like this